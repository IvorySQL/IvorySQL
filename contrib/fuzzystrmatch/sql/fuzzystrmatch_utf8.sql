/*
 * This test must be run in a database with UTF-8 encoding,
 * because other encodings don't support all the characters used.
 */

SELECT getdatabaseencoding() <> 'UTF8'
       AS skip_test \gset
\if :skip_test
\quit
\endif

set client_encoding = utf8;

-- CREATE EXTENSION IF NOT EXISTS fuzzystrmatch;

-- Accents
SELECT daitch_mokotoff('Müller');
SELECT daitch_mokotoff('Schäfer');
SELECT daitch_mokotoff('Straßburg');
SELECT daitch_mokotoff('Éregon');

-- Special characters added at https://www.jewishgen.org/InfoFiles/Soundex.html
SELECT daitch_mokotoff('gąszczu');
SELECT daitch_mokotoff('brzęczy');
SELECT daitch_mokotoff('ţamas');
SELECT daitch_mokotoff('țamas');
