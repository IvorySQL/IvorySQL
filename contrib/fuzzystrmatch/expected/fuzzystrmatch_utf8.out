/*
 * This test must be run in a database with UTF-8 encoding,
 * because other encodings don't support all the characters used.
 */
SELECT getdatabaseencoding() <> 'UTF8'
       AS skip_test \gset
\if :skip_test
\quit
\endif
set client_encoding = utf8;
-- CREATE EXTENSION IF NOT EXISTS fuzzystrmatch;
-- Accents
SELECT daitch_mokotoff('Müller');
 daitch_mokotoff 
-----------------
 {689000}
(1 row)

SELECT daitch_mokotoff('Schäfer');
 daitch_mokotoff 
-----------------
 {479000}
(1 row)

SELECT daitch_mokotoff('Straßburg');
 daitch_mokotoff 
-----------------
 {294795}
(1 row)

SELECT daitch_mokotoff('Éregon');
 daitch_mokotoff 
-----------------
 {095600}
(1 row)

-- Special characters added at https://www.jewishgen.org/InfoFiles/Soundex.html
SELECT daitch_mokotoff('gąszczu');
 daitch_mokotoff 
-----------------
 {564000,540000}
(1 row)

SELECT daitch_mokotoff('brzęczy');
        daitch_mokotoff        
-------------------------------
 {794640,794400,746400,744000}
(1 row)

SELECT daitch_mokotoff('ţamas');
 daitch_mokotoff 
-----------------
 {364000,464000}
(1 row)

SELECT daitch_mokotoff('țamas');
 daitch_mokotoff 
-----------------
 {364000,464000}
(1 row)

